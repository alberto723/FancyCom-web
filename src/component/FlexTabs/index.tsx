import React, { useState } from 'react';
import { SortableContainer, SortableElement } from 'react-sortable-hoc';
import arrayMove from 'array-move';
import { makeStyles } from '@material-ui/core/styles';
// import './style.css';
import Dialog from '@material-ui/core/Dialog';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import ExpandMoreOutlinedIcon from '@material-ui/icons/ExpandMoreOutlined';
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';

interface TabProps {
  id: string;
  name: string;
}

interface TabActionProps {
  label: string;
  onSelect: object;
}

export interface FlexTabsProps {
  name?: string;
  tabs: TabProps[];
  enableTabDropdowns?: boolean;
  enableContextMenu?: string;
  tabActions?: TabActionProps[];
  canRename?: boolean;
  canDrag?: boolean;
  overflow?: string;
  canAdd: boolean;
  componentWrapper?: object;
  // style: ViewStyle;
  tabComponent?: object;
  vertical?: boolean;
}

export const FlexTabs: React.FC<FlexTabsProps> = ({
  name,
  tabs,
  enableTabDropdowns,
  enableContextMenu,
  tabActions,
  canRename,
  canDrag,
  overflow,
  canAdd,
  componentWrapper,
  // style,
  tabComponent,
  vertical
}) => {
  const styles: any = useStyles();
  const [tabsData, setTabsData] = useState(tabs);
  const [nSelItem, setNSelItem] = useState(tabs[0]);
  const [modalVisible, setModalVisible] = useState(false);
  const [editModalVisible, setEditModalVisible] = useState(false);
  const [tabTitle, setTabTitle] = useState('');
  const [bEditOrAdd, setEditOrAdd] = useState(false);
  // console.log('enableTabDropdowns', enableTabDropdowns);
  // console.log('canRename', canRename);
  // console.log('canAdd', canAdd);

  const SortableItem = SortableElement(({ item }: { item: TabProps }) => (
    <div
      onClick={() => setNSelItem(item)}
      className={styles.tabBtn}
      id={nSelItem.id === item.id ? 'isActive' : 'notActive'}
      style={{ borderBottomColor: nSelItem.id === item.id ? '#008DD8' : 'white' }}>
      <p
        className={styles.tabTitle}
        id={item.name}
        style={{ color: nSelItem.id === item.id ? '#008DD8' : '#607380' }}>
        {item.name}
      </p>
      {enableTabDropdowns && (
        <div
          onClick={() => {
            setTabTitle(item.name);
            setModalVisible(true);
          }}
          style={{ width: 20 }}>
          {nSelItem.id === item.id && (
            <ExpandMoreOutlinedIcon style={{ fontSize: 20, color: '#008DD8', marginLeft: 5 }} />
          )}
        </div>
      )}
    </div>
  ));

  const SortableList = SortableContainer(({ items }: { items: TabProps[] }) => {
    return (
      <div style={{ display: 'flex' }}>
        {items.map((value, index) => (
          <SortableItem key={`value-${index}`} index={index} item={value} />
        ))}
      </div>
    );
  });

  const onSortEnd = ({ oldIndex, newIndex }: { oldIndex: number; newIndex: number }) => {
    setTabsData(arrayMove(tabsData, oldIndex, newIndex));
  };

  const guidGenerator = () => {
    var S4 = function () {
      return (((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1);
    };
    return S4() + S4() + '-' + S4() + '-' + S4() + '-' + S4() + '-' + S4() + S4() + S4();
  };

  const renderContextModal = (
    <Dialog open={modalVisible}>
      <DialogContent>
        <div className={styles.centeredView}>
          <div className={styles.modalView}>
            {tabActions &&
              tabActions.map((item, index) => (
                <div
                  key={index.toString()}
                  className={styles.openButton}
                  onClick={() => {
                    setModalVisible(!modalVisible);
                    if (item.label === 'Rename') {
                      if (canRename) {
                        setTimeout(() => {
                          // item.onSelect();
                          setEditOrAdd(false);
                          setEditModalVisible(true);
                        }, 1000);
                      }
                    } else if (item.label === 'Duplicate') {
                      const tmpTabsData = tabsData;
                      var tmp = { id: guidGenerator(), name: nSelItem.name };
                      tmpTabsData.push(tmp);
                      setTabsData(tmpTabsData);
                    } else if (item.label === 'Delete') {
                      const tmpTabsData = tabsData;
                      if (tmpTabsData.length > 1) {
                        var index = tmpTabsData.indexOf(nSelItem);
                        tmpTabsData.splice(index, 1);
                        setNSelItem(tmpTabsData[0]);
                        setTabsData(tmpTabsData);
                      } else {
                        alert('You can remove the last menu');
                      }
                    }
                  }}>
                  <p className={styles.textStyle}>{item.label}</p>
                </div>
              ))}
          </div>
        </div>
      </DialogContent>
    </Dialog>
  );

  const onChangeTitleText = (event: any) => {
    setTabTitle(event.target.value);
  };

  const renderEditTabTitle = (
    <Dialog open={editModalVisible}>
      <DialogTitle id='alert-dialog-title'>Tab Title</DialogTitle>
      <div className={styles.centeredView}>
        <div className={styles.modalView}>
          <textarea
            style={{
              width: '100%',
              height: 40,
              borderColor: 'grey',
              borderWidth: 1,
              borderRadius: 10
            }}
            onChange={onChangeTitleText}
            value={tabTitle}
            placeholder='Tab Title'
          />
          <div
            className={styles.openButton}
            onClick={() => {
              const tmpTabsData = tabsData;
              if (tabTitle !== '') {
                if (bEditOrAdd && canAdd) {
                  var tmp = { id: guidGenerator(), name: tabTitle };
                  tmpTabsData.push(tmp);
                } else if (!bEditOrAdd && canRename) {
                  for (var i = 0; i < tabsData.length; i++) {
                    if (tabsData[i].id === nSelItem.id) {
                      tmpTabsData[i].name = tabTitle;
                      break;
                    }
                  }
                }
              }
              setTabsData(tmpTabsData);
              setEditModalVisible(!editModalVisible);
            }}>
            <p className={styles.textStyle}>OK</p>
          </div>
        </div>
      </div>
    </Dialog>
  );

  const renderNormalItem = () => {
    return (
      <div style={{ display: 'flex' }}>
        {tabsData.map((item, index) => (
          <div
            key={index.toString()}
            onClick={() => setNSelItem(item)}
            className={styles.tabBtn}
            style={{ borderBottomColor: nSelItem.id === item.id ? '#008DD8' : 'white' }}>
            <p
              style={{
                color: nSelItem.id === item.id ? '#008DD8' : '#607380',
                marginTop: 2,
                marginBottom: 2
              }}>
              {item.name}
            </p>
            {enableTabDropdowns && (
              <div
                onClick={() => {
                  setTabTitle(item.name);
                  setModalVisible(true);
                }}
                style={{ width: 20 }}>
                {nSelItem.id === item.id && (
                  <ExpandMoreOutlinedIcon
                    style={{ fontSize: 20, color: '#008DD8', marginLeft: 5 }}
                  />
                )}
              </div>
            )}
          </div>
        ))}
      </div>
    );
  };

  return (
    <div className={styles.screen}>
      <div className={styles.flexTabContainer}>
        <div
          style={{
            alignItems: 'center',
            justifyContent: 'center',
            paddingRight: 15,
            paddingLeft: 15,
            flex: 1
          }}>
          {canDrag ? (
            <SortableList
              items={tabsData}
              axis='x'
              lockAxis='x'
              onSortEnd={onSortEnd}
              distance={1}
            />
          ) : (
            renderNormalItem()
          )}
        </div>
        <div
          onClick={() => {
            setTabTitle('');
            setEditOrAdd(true);
            setEditModalVisible(true);
          }}
          style={{ marginLeft: 5, marginRight: 5 }}>
          <MoreHorizIcon style={{ fontSize: 25, color: '#607380' }} />
        </div>
      </div>
      {modalVisible && renderContextModal}
      {editModalVisible && renderEditTabTitle}
    </div>
  );
};

interface wrapperProps {
  children: React.ReactNode;
}

const MyComponentWrapper: React.FC<wrapperProps> = ({ children }) => {
  return (
    <div
    // style={{ borderWidth: 2, borderColor: 'green' }}
    >
      {children}
    </div>
  );
};

const MyTabComponent: React.FC<wrapperProps> = ({ children }) => {
  return (
    <div
    // style={{ borderWidth: 2, borderColor: 'red' }}
    >
      {children}
    </div>
  );
};

FlexTabs.defaultProps = {
  enableTabDropdowns: true,
  canAdd: true,
  canRename: true,
  canDrag: true,
  componentWrapper: MyComponentWrapper,
  tabComponent: MyTabComponent
};

const useStyles = makeStyles(() => ({
  screen: {
    flex: 1,
    backgroundColor: 'white',
  },
  flexTabContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderWidth: '0px',
    borderBottomColor: 'grey',
    borderBottomWidth: '1px',
    borderTopColor: 'grey',
    borderTopWidth: '1px',
    marginTop: '5px',
    marginBottom: '5px',
    borderStyle: 'solid',
    width: '100%',
  },
  tabBtn: {
    display: 'flex',
    paddingLeft: '10px',
    paddingRight: '10px',
    marginLeft: '10px',
    marginRight: '10px',
    borderWidth: '0px',
    borderBottomWidth: '2px',
    borderStyle: 'solid',
    flexDirection: 'row',
    alignItems: 'center',
  },
  tabTitle: {
    marginTop: '2px',
    marginBottom: '2px',
  },
  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    minWidth: '300px',
  },
  modalView: {
    backgroundColor: 'white',
    borderRadius: '20px',
    padding: '35px',
    alignItems: 'center',
  },
  openButton: {
    backgroundColor: '#2196F3',
    borderRadius: '20px',
    paddingLeft: '10px',
    paddingRight: '10px',
    paddingTop: '2px',
    paddingBottom: '2px',
    marginTop: '15px',
    marginBottom: '15px',
  },
  textStyle: {
    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center',
    marginTop: '10px',
    marginBottom: '10px',
  },
}));
