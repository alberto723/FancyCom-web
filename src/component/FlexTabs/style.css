.screen {
  flex: 1;
  background-color: white;
}
.flexTabContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  border-width: 0px;
  border-bottom-color: grey;
  border-bottom-width: 1px;
  border-top-color: grey;
  border-top-width: 1px;
  margin-top: 5px;
  margin-bottom: 5px;
  border-style: solid;
  width: 100%;
}
.tabBtn {
  display: flex;
  padding-left: 10px;
  padding-right: 10px;
  margin-left: 10px;
  margin-right: 10px;
  border-width: 0px;
  border-bottom-width: 2px;
  border-style: solid;
  flex-direction: row;
  align-items: center;
}
.centeredView {
  flex: 1;
  justify-content: center;
  align-items: center;
  min-width: 300px;
}
.modalView {
  background-color: white;
  border-radius: 20px;
  padding: 35px;
  align-items: center;
}
.openButton {
  background-color: #2196F3;
  border-radius: 20px;
  padding-left: 10px;
  padding-right: 10px;
  padding-top: 2px;
  padding-bottom: 2px;
  margin-top: 15px;
  margin-bottom: 15px;
}
.textStyle {
  color: white;
  font-weight: bold;
  text-align: center;
  margin-top: 10px;
  margin-bottom: 10px;
}
