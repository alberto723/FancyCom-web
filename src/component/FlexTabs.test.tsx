// Canvas.test.js
import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

import { FlexTabs } from './FlexTabs';

const tabItems = [
  { id: '1', name: 'Builder' },
  { id: '2', name: 'Calendar' },
  { id: '3', name: 'Insights' },
  { id: '4', name: 'Moments' },
  { id: '5', name: 'About' },
  { id: '6', name: 'Terminal' }
];

const tabActionItems = [
  { label: 'Rename', onSelect: () => console.log('onSelect clicked') },
  { label: 'Duplicate', onSelect: () => console.log('onDuplicate clicked') },
  { label: 'Delete', onSelect: () => console.log('onDelete clicked') }
];

test('Should be rendered in the dom', () => {
  const component = renderer.create(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should render a list of tabs in the dom that number the same as the number of tabs in the tabs array prop', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should render each tab with a caption that matches the tab item associated with that tab in the tabs array prop', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );
  // container.querySelector('.drawerButton')?.click();

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should render each tab with an icon (if one is present) that matches the tab item associated with that tab in the tabs array prop', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );

  expect(container.querySelectorAll('.makeStyles-tabBtn-3')[0].id.includes('isActive')).toBe(true);
});

test('Should render the tab as "active" it the tab in the tabs array prop has the isActive property set to true', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );

  expect(container.querySelectorAll('.makeStyles-tabBtn-3')[0].id.includes('isActive')).toBe(true);
});

test('Should render a dropdown of actions where the number of actions match the number of items in the tabActions prop', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );
  // container.querySelector('.drawerButton')?.click();

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should render a dropdown of actions on each tab if the enableTabDropdowns prop is set to true', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should render a dropdown of actions that are activated on right-click if the enableContextMenu prop is set to true', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should not permit renaming if the canRename prop is false', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );
  // container.querySelector('.drawerButton')?.click();

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should not permit re-ordering if the canDrag prop is false', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should not permit addingnew tabs if the canAdd prop is false', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should render the tabs as a carousel when the number of tabs exceed the number that can fit on the screen', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );
  // container.querySelector('.drawerButton')?.click();

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should render the tabs normally with the overflow tabs hidden if the overflow prop is set to HIDDEN', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );
  // container.querySelector('.drawerButton')?.click();

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should render the component in the componentWrapper prop if one is provided', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );
  // container.querySelector('.drawerButton')?.click();

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should render the component in the tabComponent prop if one is provided', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );
  // container.querySelector('.drawerButton')?.click();

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should render the same number of tabComponent props as the amount of tabs in the tabs array prop', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );
  // container.querySelector('.drawerButton')?.click();

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});

test('Should render the tabs as vertical if the vertical prop is set to true', () => {
  const { queryByLabelText, getByText, container } = render(
    <FlexTabs
      name='flextab'
      tabs={tabItems}
      enableTabDropdowns={true}
      tabActions={tabActionItems}
      canRename={true}
      canAdd={true}
      canDrag={true}
    />
  );

  expect(container.querySelectorAll('.makeStyles-tabBtn-3').length).toBe(tabItems.length);
});
