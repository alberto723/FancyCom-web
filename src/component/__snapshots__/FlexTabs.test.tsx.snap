// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be rendered in the dom 1`] = `
<div
  className="makeStyles-screen-1"
>
  <div
    className="makeStyles-flexTabContainer-2"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          className="makeStyles-tabBtn-3"
          id="isActive"
          onClick={[Function]}
          style={
            Object {
              "borderBottomColor": "#008DD8",
            }
          }
        >
          <p
            className="makeStyles-tabTitle-4"
            id="Builder"
            style={
              Object {
                "color": "#008DD8",
              }
            }
          >
            Builder
          </p>
          <div
            onClick={[Function]}
            style={
              Object {
                "width": 20,
              }
            }
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              style={
                Object {
                  "color": "#008DD8",
                  "fontSize": 20,
                  "marginLeft": 5,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41z"
              />
            </svg>
          </div>
        </div>
        <div
          className="makeStyles-tabBtn-3"
          id="notActive"
          onClick={[Function]}
          style={
            Object {
              "borderBottomColor": "white",
            }
          }
        >
          <p
            className="makeStyles-tabTitle-4"
            id="Calendar"
            style={
              Object {
                "color": "#607380",
              }
            }
          >
            Calendar
          </p>
          <div
            onClick={[Function]}
            style={
              Object {
                "width": 20,
              }
            }
          />
        </div>
        <div
          className="makeStyles-tabBtn-3"
          id="notActive"
          onClick={[Function]}
          style={
            Object {
              "borderBottomColor": "white",
            }
          }
        >
          <p
            className="makeStyles-tabTitle-4"
            id="Insights"
            style={
              Object {
                "color": "#607380",
              }
            }
          >
            Insights
          </p>
          <div
            onClick={[Function]}
            style={
              Object {
                "width": 20,
              }
            }
          />
        </div>
        <div
          className="makeStyles-tabBtn-3"
          id="notActive"
          onClick={[Function]}
          style={
            Object {
              "borderBottomColor": "white",
            }
          }
        >
          <p
            className="makeStyles-tabTitle-4"
            id="Moments"
            style={
              Object {
                "color": "#607380",
              }
            }
          >
            Moments
          </p>
          <div
            onClick={[Function]}
            style={
              Object {
                "width": 20,
              }
            }
          />
        </div>
        <div
          className="makeStyles-tabBtn-3"
          id="notActive"
          onClick={[Function]}
          style={
            Object {
              "borderBottomColor": "white",
            }
          }
        >
          <p
            className="makeStyles-tabTitle-4"
            id="About"
            style={
              Object {
                "color": "#607380",
              }
            }
          >
            About
          </p>
          <div
            onClick={[Function]}
            style={
              Object {
                "width": 20,
              }
            }
          />
        </div>
        <div
          className="makeStyles-tabBtn-3"
          id="notActive"
          onClick={[Function]}
          style={
            Object {
              "borderBottomColor": "white",
            }
          }
        >
          <p
            className="makeStyles-tabTitle-4"
            id="Terminal"
            style={
              Object {
                "color": "#607380",
              }
            }
          >
            Terminal
          </p>
          <div
            onClick={[Function]}
            style={
              Object {
                "width": 20,
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "marginLeft": 5,
          "marginRight": 5,
        }
      }
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#607380",
            "fontSize": 25,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </div>
  </div>
</div>
`;
